src/esbmc/bmc.cpp

```diff
@@ -731,104 +732,118 @@ smt_convt::resultt bmct::multi_property_check(
         {
-          const std::lock_guard<std::mutex> lock(result_mutex);
-          // Check if someone else found the solution
-          if(fail_fast && final_result == smt_convt::P_SATISFIABLE)
-          {
-            log_status(
-              "Found solution for VCC. But, other thread found it first.");
-            throw 0;
-          }
-          goto_tracet goto_trace;
-          build_goto_trace(local_eq, runtime_solver, goto_trace, false);
-          // TODO: Replace this with a test-case for coverage!
-          std::string output_file = options.get_option("cex-output");
-          if(output_file != "")
-          {
-            std::ofstream out(fmt::format("{}-{}", output_file, ce_counter++));
-            show_goto_trace(out, ns, goto_trace);
-          }
-          std::ostringstream oss;
-          log_fail("\n[Counterexample]\n");
-          show_goto_trace(oss, ns, goto_trace);
-          log_result("{}", oss.str());
-          final_result = result;
-
-          // collect the tracked instrumentation which is verified failed
-          // we assume it always works in multi-property checking mode
-          if(
-            options.get_bool_option("goto-coverage") ||
-            options.get_bool_option("make-assert-false") ||
-            options.get_bool_option("add-false-assert"))
+          log_status(
+            "Found solution for VCC. But, other thread found it first.");
+          throw 0;
+        }
+        goto_tracet goto_trace;
+        build_goto_trace(local_eq, runtime_solver, goto_trace, false);
+        // TODO: Replace this with a test-case for coverage!
+        std::string output_file = options.get_option("cex-output");
+        if(output_file != "")
+        {
+          std::ofstream out(fmt::format("{}-{}", output_file, ce_counter++));
+          show_goto_trace(out, ns, goto_trace);
+        }
+        std::ostringstream oss;
+        log_fail("\n[Counterexample]\n");
+        show_goto_trace(oss, ns, goto_trace);
+        log_result("{}", oss.str());
+        final_result = result;
+
+        // collect the tracked instrumentation which is verified failed
+        // we assume it always works in multi-property checking mode
+        if(
+          options.get_bool_option("goto-coverage") ||
+          options.get_bool_option("make-assert-false") ||
+          options.get_bool_option("add-false-assert"))
+        {
+          if(claim.claim_msg.find("Instrumentation") != std::string::npos)
           {
-            if(claim.claim_msg.find("Instrumentation") != std::string::npos)
+            // E.g.
+            // "Claim 1: Instrumentation ASSERT(0) Added"
+            // "Claim 2: Instrumentation ASSERT(0) Converted"
+            if(!reached_claims.count(claim.claim_msg))
+            {
+              // avoid double counting the claims in loop/func_call
               tracked_instrument++;
+              reached_claims.insert(claim.claim_msg);
```

Comment:
This pattern of a set.find() (or set.count()) condition followed by set.insert() is inefficient. Suggest to use `if(auto [it, ins] = set.try_emplace(...); ins)` instead.

Reply:
Changed as request.